containers/base/Dockerfile

```diff
@@ -23,153 +23,161 @@ MAINTAINER Google Cloud DataLab
+ENV PATH $PATH:/tools/node/bin:/tools/google-cloud-sdk/bin:$DATALAB_CONDA_DIR/bin
+
+ENV PYTHON_2_ENV py2env
+ENV PYTHON_3_ENV py3env
+
+# Needed to run "source" for switching Conda environments.
+SHELL ["/bin/bash", "-c"]
 
 # Setup OS and core packages
 RUN echo "deb-src http://ftp.us.debian.org/debian testing main" >> /etc/apt/sources.list && \
     apt-get update -y && \
     apt-get install -y -q debian-archive-keyring debian-keyring && \
     apt-get update -y && \
-    apt-get install --no-install-recommends -y -q python unzip ca-certificates build-essential \
-    libatlas-base-dev liblapack-dev gfortran libpng-dev libfreetype6-dev libxft-dev libxml2-dev \
-    libsnappy-dev python-dev python-setuptools python-zmq openssh-client wget curl git \
-    pkg-config zip && \
-    easy_install pip && \
+    apt-get install --no-install-recommends -y -q \
+        build-essential \
+        ca-certificates \
+        curl \
+        git \
+        locales \
+        openssh-client \
+        pkg-config \
+        unzip \
+        wget \
+        zip && \
     mkdir -p /tools && \
-
 # Save GPL source packages
     mkdir -p /srcs && \
     cd /srcs && \
     apt-get source -d wget git python-zmq ca-certificates pkg-config libpng-dev && \
-    wget --progress=dot:mega https://mirrors.kernel.org/gnu/gcc/gcc-4.9.2/gcc-4.9.2.tar.bz2 && \
     cd / && \
+# Set our locale to en_US.UTF-8.
+    locale-gen en_US.UTF-8 && \
+    update-locale LANG=en_US.UTF-8 && \
+# Install MiniConda, set up Python 2.7 Conda env, and install packages.
+    wget --quiet -O ~/miniconda.sh \
+    http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh && \
+    chmod +x ~/miniconda.sh && \
+    ~/miniconda.sh -b -f -p $DATALAB_CONDA_DIR && \
+    rm ~/miniconda.sh && \
+    conda config --system --add channels conda-forge && \
+    conda config --system --set show_channel_urls true && \
+    conda update --all --quiet --yes && \
+    conda create --yes --quiet --name $PYTHON_2_ENV python=2.7 \
+        crcmod==1.7 \
+        future==0.16.0 \
+        futures==3.2.0 \
+        google-api-python-client==1.6.2 \
+        httplib2==0.10.3 \
+        h5py==2.7.1 \
+        ipykernel==4.5.2 \
+        ipywidgets==6.0.0 \
+        jinja2==2.8 \
+        jsonschema==2.6.0 \
+        matplotlib==2.1.2 \
+        mock==2.0.0 \
+        nltk==3.2.1 \
+        numpy==1.12.1 \
+        oauth2client==2.2.0 \
+        pandas-gbq==0.3.0 \
+        pandas==0.22.0 \
+        pandocfilters==1.4.2 \
+        pillow==5.0.0 \
+        plotly==1.12.5 \
+        psutil==4.3.0 \
+        pygments==2.1.3 \
+        python-dateutil==2.5.0 \
+        python-snappy==0.5.1 \
+        pytz==2016.7 \
+        pyzmq==16.0.2 \
+        requests==2.18.4 \
+        scikit-image==0.13.0 \
+        scikit-learn==0.18.2 \
+        scipy==1.0.0 \
+        seaborn==0.7.0 \
+        six==1.11.0 \
+        statsmodels==0.6.1 \
+        sympy==0.7.6.1 \
+        tensorflow==1.5.0 \
```

Comment:
Just realized an issue: Does this work with the GPU container?

You didn't change Dockerfile.gpu at all...